src/languages/php.js

```diff
@@ -18,18 +19,50 @@ export default function(hljs) {
       { begin: /\?>/ } // end php tag
     ]
   };
+  var SUBST ={
+    className: 'subst',
+    variants: [
+      {begin: /\$\w+/},
+      {begin: /\{\$/, end: /\}/}
+    ]
+  };
+  var SINGLE_QUOTED = hljs.inherit(
+    hljs.APOS_STRING_MODE,
+    {
+      illegal: null,
+      variants: [{
+        begin: 'b\'', end: '\''
+      }]
+    }
+  );
+  var DOUBLE_QUOTED = hljs.inherit(
+    hljs.QUOTE_STRING_MODE,
+    {
+      illegal: null,
+      variants: [{
+        begin: 'b"', end: '"'
+      }],
+      contains: [hljs.BACKSLASH_ESCAPE, SUBST]
+    }
+  );
+  var HEREDOC = {
+    begin: /<<<[ \t]*(?<marker>\w+)\n(?:.*\n)*?^[ \t]*\k<marker>\b/,
+    returnBegin: true,
+    contains: [
+      { begin: /<<<[ \t]*/ },
```

Comment:
I don't think se need any of this... move the <<< into the BEGIN match...  we shouldn't only need a simple being/end match here (inside the END_SAME_AS_BEGIN) helper, no?

Reply:
Ie, we don't need to find the end first...

Reply:
right, I missed that copying from ruby.js